### PR TITLE
Make nuke ops scale with station population

### PR DESCRIPTION
**Pull request by neersighted**
The current min is 4, max is 10, and scaling factor (population / factor) is 10.

Also fix up some shitcode while I was in there.

### CODE REVIEW

**Comment by KorPhaeron** [comment]
@neersighted 

Keep in mind every nuke op comes with 20tc of their own, so you're giving them a ton of extra gear as well (maybe disable normal TC scaling?)

**Comment by ChangelingRain** [comment]
This won't spawn additional suits, TC stations, pinpointers, or lockers, though.

**Comment by neersighted** [comment]
Nope, this is just free reinforcements. The other things will be have to be purchased/shared by Ops, its not just a net increase.

**Comment by KorPhaeron** [comment]
@neersighted 

The default nuke op spawns with a 20tc radio, so yes you are giving them more TC unless you changed the default outfit

**Comment by WJohn** [comment]
More buffs :+1: 
I always like seeing the antags kicking the shit out of the crew (except with blobs because blobbernaut spam isn't interesting).

**Comment by Incoming5643** [comment]
Current Practical scaling

Under 20 players: Verboten
20 to 40 players: 16 to 36 crew and 4 ops
41 to 50 players: 36 to 45 crew and 5 ops
51 to 60 players: 45 to 54 crew and 6 ops
61 to 70 players: 54 to 63 crew and 7 ops
71 to 80 players: 63 to 72 crew and 8 ops
81 to 90 players: 72 to 81 crew and 9 ops
Over 90 players: 81+ crew and 10 ops

I feel like this is slanted pretty damn hard towards ops. Especially if they get 20 TC radios.

Also it doesn't take in to consideration how full/empty head roles are, but unfortunately that'd be very had to check the way game mode code currently is.

**Comment by KorPhaeron** [comment]
Apparently it is actually 30tc radios

**Comment by neersighted** [comment]
The numbers are all up for change, I have Jo idea what they should be.

**Comment by Incoming5643** [comment]
I added TC counts for your scaling to my post. 

Look at /obj/machinery/computer/telecrystals/boss/proc/getDangerous() to see how bonus TC is added.

**Comment by neersighted** [comment]
I plan to kill that in a minute. 20tc radios might also be a good idea.

**Comment by Lumipharon** [comment]
reduce the minimm number.
4 ops vs 20 crew is still guaranteed victory aside from massive fluke.

**Comment by neersighted** [comment]
Someone give me a comprehensive list of:
* Min Ops
* Max Ops
* Ops Factor (population / Ops Factor)
* Bonus Factor (population * Bonus Factor)
* TCs per Op

I don't want to fight over balance, I'm just offering up the code.

**Comment by Lumipharon** [comment]
Honestly do we even need a min and max?
2ops v 20 crew isn't underpowered considering there is unlikely to be much (or any) sec.
1 op will only happen due to badmin forcing round type on evenlower pop.

For the max limit, if we have more then 100 people on the server, then the ops need all the help they can get, as they'll basically run out of ammo trying to stop an unending tide of validhunters.

As for the other stuff, honestly it would be easiest to just test merge is and see how it works out at different pop levels, then trying to theorycraft a perfect fit before hand.

File: code/game/gamemodes/nuclear/nuclear.dm [hunk 1]
```diff
@@ -23,34 +25,22 @@
...
-			var/mob/living/carbon/human/human = synd_mind.current
-			if(human.dna && human.dna.species.dangerous_existence)
-				human.set_species(/datum/species/human)
-
+	var/agent_count = Clamp(num_players() / agent_factor, min_agents, max_agents)
+
+	var/agents_created = 0
+	while(agents_created < agent_count && antag_candidates.len)
+		syndicates += pick_n_take(antag_candidates)
+		agents_created++
```
Comment (duncathan) [inline code/game/gamemodes/nuclear/nuclear.dm]: `for(var/agents_created = 0; agents_created < agent_count && antag_candidates.len; agents_created++)`

**Comment by duncathan** [comment]
@KorPhaeron your call

**Comment by KorPhaeron** [comment]
Gonna test merge this for some time period first

**Comment by Cheridan** [comment]
I'd much much rather see TC scaling adjusted rather than adding more operatives. Adding more just makes it less tactical and harder to cooperate. We already have Rev, Gang, Cult for mass zerg combat.

**Comment by Lumipharon** [comment]
When you have 70 people to fight, there is nothing tactical about it, unless you're being some shitter stealth ops, which as always, has only ever required a single guy to do. Also, more tc = being able to just buy more ops anyway, so eh?

This PR is also far more significantly about having LESS ops at lower levels, where it's just a completely boring steamroll.

**Comment by neersighted** [comment]
Thanks for summing my thoughts up @Lumipharon

**Comment by KorPhaeron** [comment]
http://ss13.eu/tgdb/tg/latest_stats.html#mode_nuclear_emergency

Latest stats

**Comment by lzimann** [comment]
Looks like nuke ops are spawning naked somehow.

**Comment by KorPhaeron** [comment]
Let me know when you fix it

**Comment by KorPhaeron** [comment]
Ping me if you want it re-opened

**Pull request closed without merging**